syntax = "proto3";

package heartbeat_system;

service HeartBeat {
  rpc RegisterNode (NodeInfo) returns (RegistrationStatus);
  rpc Heartbeat (HeartbeatRequest) returns (HeartbeatResponse);
	rpc GetClusterStatus (ClientInfo) returns (ClusterStatus);
}

message ClientInfo{
	repeated int32 nodes_info = 1;
}

message ClusterStatus{
  repeated string new_nodes = 1;
  repeated int32 removed_nodes = 2;
}

message NodeInfo {
  string node_id = 1;
  string address = 2;
}

message RegistrationStatus {
  bool success = 1;
  int64 broker_id = 2;
  string message = 3;
}

message HeartbeatRequest {
  string node_id = 1;
}

message HeartbeatResponse {
  bool alive = 1;
}

